recipe-server/normandy/recipes/api/v2/shield_identicon.py

```diff
@@ -0,0 +1,81 @@
+from django.template.loader import render_to_string
+
+
+def pick_pair(genome, base_color):
+    good_pairs = [
+        color for color in genome.colors if base_color.contrasts_well(color)]
+    assert len(good_pairs), 'no colors satisfy the luminance requirement'
+    return genome.choice(good_pairs)
+
+
+def single_color(genome, context):
+    context['treatment'] = 'SingleColor'
+    return context
+
+
+def two_color(genome, context):
+    context['treatment'] = 'TwoColor'
+    areas = [
+        'top',
+        'bottom',
+        'left',
+        'right',
+        'topLeft',
+        'topRight',
+        'bottomLeft',
+        'bottomRight',
+    ]
+    area = genome.choice(areas)
+    rotations = {
+        'top': 0,
+        'topRight': 45,
+        'right': 90,
+        'bottomRight': 135,
+        'bottom': 180,
+        'bottomLeft': 225,
+        'left': 270,
+        'topLeft': 315,
+    }
+    context['transform'] = f'scale(100) rotate({rotations[area]} 0.5,0.5)'
+    return context
+
+
+def stripes(genome, context):
+    context['treatment'] = 'Stripes'
+    count = genome.randint(1, 4)
+    padding = genome.uniform(0.1, 0.4)
+    directions = ['vertical', 'horizontal', 'diagonal1', 'diagonal2']
+    direction = genome.choice(directions)
+    stride = (1 - 2 * padding) / (2 * count + 1)
+
+    context['stripe_x_list'] = [
+        padding + stride * (2 * i + 1) for i in range(count)]
+    context['stride'] = stride
+    rotations = {
+        'vertical': 0,
+        'diagonal1': 45,
+        'horizontal': 90,
+        'diagonal2': 135
+    }
+    context['transform'] = f'scale(100) rotate({rotations[direction]} 0.5,0.5)'
+    return context
+
+
+def generate_svg(genome):
+    treatments = [
+        {'function': single_color, 'weight': 1},
+        {'function': two_color, 'weight': 4},
+        {'function': stripes, 'weight': 6},
+    ]
+    update_context_with_treatment = genome.weighted_choice(treatments)['function']
+    emoji = genome.emoji()
```

Comment:
Nit: No need to store this in a variable, you can just call it directly in the dict below.